### PR TITLE
update wasm tools dep to produce single file output wit

### PR DESCRIPTION
**Pull request by karthik2804**
Updates wasm-tools to be able to handle multiple packages in single `wit` file. This also changes the output format to a single file.

It also changes the order in which `wit` files are merged - this is a hack to temporarily get around `wit merge` not being commutative in `wasm-tools`. This relies on the fact that this is mostly only used for JS projects where the entire `wit` is encoded without minimization.

### CODE REVIEW

File: crates/knitwit/src/lib.rs [hunk 1]
```diff
@@ -49,12 +44,13 @@ impl Guest for KnitWit {
...
         merge_worlds(&mut resolve, base_world, worlds).map_err(|e| e.to_string())?;
 
-        resolve_to_readable_wit(resolve, id, output_dir).map_err(|e| e.to_string())
+        resolve_to_readable_wit(&resolve, id).map_err(|e| e.to_string())
     }
 }
 
 fn merge_wits(resolve: &mut Resolve, wit_paths: Vec<String>) -> anyhow::Result<()> {
     for path in wit_paths {
+        println!("Merging wit: {}", path);
```
Comment (radu-matei) [inline crates/knitwit/src/lib.rs]: Curious if this is user actionable / useful?

**Review by radu-matei — APPROVED** [review]

File: crates/knitwit/src/lib.rs (hunk 1 above)
Comment (karthik2804) [reply crates/knitwit/src/lib.rs]: Good catch! That comes from my debugging efforts.

**Comment by karthik2804** [comment]
I just realized I should also delete the `dist`, `lib` and `obj` folders.

**Comment by karthik2804** [comment]
It was somehow merged to main but the PR stuck around - closing

**Pull request closed without merging**